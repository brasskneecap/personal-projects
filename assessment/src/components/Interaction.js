

const Interaction = ({onClickHandler, component, typeCount, typeText}) => {
    return (
        <div className="interaction-content">
            <div className="interaction-icon"  onClick={onClickHandler}>
                {component}
            </div>
            <span className="count-display">{typeCount}</span><span className="type-display">{typeText}</span>
        </div>
    )
}

export default Interaction