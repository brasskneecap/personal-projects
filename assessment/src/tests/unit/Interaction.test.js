import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Interaction from '../../components/Interaction';
import CommentSVG from '../../components/svg-components/CommentSVG.js'

describe('Interaction', () => {
    const wrapper = new ShallowRenderer()

    const mockHandler = () => {
        console.log('test')
    }
    
    wrapper.render(
    <Interaction onClickHandler={mockHandler} component={<CommentSVG />} typeCount="1" typeText="test" />
    );
    
    let componentRendered = wrapper.getRenderOutput();
    
    it('Interaction Snapshot', () => {
        expect(componentRendered).toMatchSnapshot();
    })

    it('Should containt commentSVG', () => {
        expect(componentRendered.props.children[0].props.children.type).toBe(CommentSVG)
    })
})