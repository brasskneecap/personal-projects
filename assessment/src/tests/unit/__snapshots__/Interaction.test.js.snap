// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interaction Interaction Snapshot 1`] = `
<div
  className="interaction-content"
>
  <div
    className="interaction-icon"
    onClick={[Function]}
  >
    <CommentSVG />
  </div>
  <span
    className="count-display"
  >
    1
  </span>
  <span
    className="type-display"
  >
    test
  </span>
</div>
`;
